packages/python-packages/apiview-copilot/tests/evals/discovery_test.py

```diff
@@ -0,0 +1,452 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+Tests for discovery functionality in APIView Copilot evals.
+"""
+
+import tempfile
+from pathlib import Path
+from unittest.mock import patch
+
+import pytest
+
+from evals._discovery import (
+    discover_targets,
+    _discover_from_path,
+    _discover_single_test,
+    _find_workflow_config
+)
+
+
+class MockWorkflowConfig:
+    def __init__(self, name="test-workflow", kind="prompt"):
+        self.name = name
+        self.kind = kind
+
+
+def test_discover_targets_empty_paths():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        tests_root = Path(temp_dir)
+        
+        with pytest.raises(ValueError, match="No workflows found"):
+            discover_targets(None, tests_root)
+
+
+def test_discover_targets_nonexistent_tests_root():
+    nonexistent = Path("/nonexistent/path")
+    
+    with pytest.raises(ValueError, match="Tests directory not found"):
+        discover_targets(None, nonexistent)
+
+
+@patch('evals._discovery.load_workflow_config')
+def test_discover_single_test_file(mock_load_config):
```

Comment:
Rather than create all of these structured via code, it would likely be cleaner and easier to create fake test hierarchy in the `tests/evals` folder (`tests/evals/paths` or something) and target those. It should simplify the code her significantly.